src/SignalR/common/Http.Connections/test/HttpConnectionDispatcherTests.cs

```diff
@@ -1166,6 +1166,18 @@ public async Task RequestToActiveConnectionIdKillsPreviousConnectionLongPolling(
                 await request1.OrTimeout();
 
                 Assert.Equal(StatusCodes.Status204NoContent, context1.Response.StatusCode);
+
+                count = 0;
+                // Wait until the second request has started internally
+                while (connection.TransportTask.IsCompleted && count < 50)
```

Comment:
Why not just await the connection.TransportTask with a timeout? Is the connection.TransportTask changing from underneath you?

Reply:
It's going to be completed from the first poll request, so we're waiting until it's been set by the second request.

Reply:
Ahh. I misread. I though this was looping while it wasn't completed, not looping while it is completed.